### DISCUSSION TITLE
TessellationShader error

### DISCUSSION THREAD

**Discussion by Ron5s** [Kode/Kha#1273] [General]
I was trying to understand the tessalation example provided but Im running into the following errors.  Also I cant log into the forum. Ive tried to login on multiple computers and web browsers.

> Executing task: node C:\Development\Kha\make.js debug-html5 --debug <

Using Kha (f0a8a458) from C:\Development\Kha
Creating Kha project.
Compiling shader 1 of 5 (test.frag.glsl).
Skipping shader 2 of 5 (test.geom.glsl). 
Skipping shader 3 of 5 (test.tesc.glsl). 
Skipping shader 4 of 5 (test.tese.glsl). 
Compiling shader 5 of 5 (test.vert.glsl).
Skipping shader 1 of 9 (clear.comp.glsl).
Compiling shader 2 of 9 (painter-colored.frag.glsl).
Compiling shader 3 of 9 (painter-colored.vert.glsl).
Compiling shader 4 of 9 (painter-image.frag.glsl).  
Compiling shader 5 of 9 (painter-image.vert.glsl).
Compiling shader 6 of 9 (painter-text.frag.glsl).
Compiling shader 7 of 9 (painter-text.vert.glsl).
Compiling shader 8 of 9 (painter-video.frag.glsl).
Compiling shader 9 of 9 (painter-video.vert.glsl).
C:\Users\Ron\Documents\Kha\Projects\TessellationShader-master\Sources/Main.hx:46: characters 38-47 : Class<kha.Shaders> has no field test_geom (Suggestion: test_vert)
C:\Users\Ron\Documents\Kha\Projects\TessellationShader-master\Sources/Main.hx:47: characters 52-61 : Class<kha.Shaders> has no field test_tese (Suggestion: test_vert)
C:\Users\Ron\Documents\Kha\Projects\TessellationShader-master\Sources/Main.hx:48: characters 49-58 : Class<kha.Shaders> has no field test_tesc (Suggestion: test_vert)

Haxe compiler error.
The terminal process "C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -Command node C:\Development\Kha\make.js debug-html5 --debug" terminated with exit code: 1.

Terminal will be reused by tasks, press any key to close it.

**Comment by RobDangerous** [comment]
Yes, the forum is breaking down for some reason. But that's ok, I want to move discussions over to GitHub anyway.
That aside, tessellation shaders do not work via html5 because WebGL still only supports vertex and fragment shaders. Also Metal does not support tessellation shaders, it uses a rather different approach for that. Also it's still problematic on Direct3D because there tessellation support is more restrictive than in GLSL. Also tessellation is usually too slow on AMD hardware to be of much use. Also mesh shaders are now changing all of that up.
To sum it up - avoid tessellation shaders unless you know exactly what you're doing. You should be able to run the example via OpenGL (and if you can find the bug in my code in Direct3D) but it's problematic if you want to do cross-platform things.

**Comment by Ron5s** [comment]
Oh ok.  Good to know.  Ill ask all my questions here from now on.

I understand, thank you for the detailed answer.  I was more interested in the geometry shader setup that's why I was running that example.  I did get it to build and run.

Thank you again,
Ron
